app/src/pages/ExploreDataPage.tsx

```diff
@@ -175,22 +175,26 @@ function CarouselMadLib(props: {
                   defaultValue={props.madLib.defaultSelections[index]}
                   value={props.madLib.activeSelections[index]}
                   onChange={(event) => {
-                    let index: number = Number(event.target.name);
-                    let updatedArray: PhraseSelections = {
+                    let phraseIndex: number = Number(event.target.name);
+                    let updatePhraseSelections: PhraseSelections = {
                       ...props.madLib.activeSelections,
                     };
-                    updatedArray[index] = Number(event.target.value);
+                    updatePhraseSelections[phraseIndex] = event.target
+                      .value as string;
                     props.setMadLib({
                       ...props.madLib,
-                      activeSelections: updatedArray,
+                      activeSelections: updatePhraseSelections,
                     });
                   }}
                 >
-                  {Object.keys(phraseSegment).map((key: string) => (
-                    <MenuItem key={key} value={Number(key)}>
-                      {phraseSegment[Number(key)]}
-                    </MenuItem>
-                  ))}
+                  {Object.entries(phraseSegment)
+                    .sort((a, b) => a[0].localeCompare(b[0]))
```

Comment:
Should this be comparing a[1] to b[1] so that it's using the names rather than the fips codes? (I think it probably doesn't matter since FIPS codes are assigned alphabetically, just thought that it made more sense)

Reply:
Then the "United States" wouldn't be at the top anymore... I was actually thinking about migrating this type so that it maintains order and we don't have to rely on sorting. I left a TODO here to go back to it...